components/ipfs/ipfs_service.cc

```diff
@@ -432,6 +435,62 @@ void IpfsService::RemovePin(const std::vector<std::string>& cids,
       GetHeaders(gurl));
 }
 
+void IpfsService::RemovePinCli(std::set<std::string> cids,
+                               BoolCallback callback) {
+  if (cids.begin() == cids.end()) {
+    std::move(callback).Run(true);
+    return;
+  }
+
+  base::FilePath path = GetIpfsExecutablePath();
+  if (path.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  base::CommandLine cmdline(path);
+  cmdline.AppendArg("pin");
+  cmdline.AppendArg("rm");
+  cmdline.AppendArg("-r=true");
+
+  cmdline.AppendArg(*(cids.begin()));
+  cids.erase(cids.begin());
+  ExecuteNodeCommand(
+      cmdline, GetDataPath(),
+      base::BindOnce(&IpfsService::OnRemovePinCli, weak_factory_.GetWeakPtr(),
+                     std::move(callback), std::move(cids)));
+}
+
```

Comment:
Should we execute the command first and delete an entry afterwards? in case the command fails for some reason

Reply:
Done